Drop unused field.

porridge opened the pull request:
**What this PR does / why we need it**:

it seems that `manifestFile.apis` is a write-only field, and can safely be removed.

**Special notes for your reviewer**:

The `Capabilities.APIVersions` field, while no longer used on its own by the Helm code directly, remains exposed to helm charts as part of the `.Capabilities` struct, which may well depend on it, so it must stay untouched.

Introducing new versions of functions with one less argument seems like an overkill, so I just replaced the argument names with an underscore.

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

porridge committed:
Drop unused field.

Signed-off-by: Marcin Owsiany <porridge@redhat.com>

robertsirc reviewed (approved):
Not sure if the maintainers plan on using this in the future or not. Comments do denote it isn't used. But this looks good to me.

porridge committed:
Remove somewhat incorrect comment.

While no longer used by the helm code, this field is part of the API exposed to helm charts as part of the .Capabilities struct.

Signed-off-by: Marcin Owsiany <marcin@redhat.com>

porridge commented:
Thanks for taking a look @robertsirc !

Removed one questionable comment line that I had added, and explained the situation a bit more in the PR description.
